### PR TITLE
feat: add lumo based theme for Keycloak provider

### LINKED ISSUE #18: Styling of Keycloak login page
Part of #17

### PR DESCRIPTION
**Pull request by DiegoCardoso**
## Description

Add a module to build a jar containing the lumo-based theme to be added to a Keycloak instance.

The theme contains a common CSS file with CSS properties and styles taken from Lumo. It would be nicer to have it injected as a npm dependency, so we wouldn't need to copy the styles, but it can be introduced as an enhancement in another ticket. 

### To test it locally:

- Build the module with `mvn package`
- Copy the `sso-kit-keycloak-theme-1.0-SNAPSHOT.jar` file from the `target` folder and paste it into the `providers` folder of your local Keycloak instance
- Start the Keycloak server
- Go to the Admin console and then "Realm settings > Themes" and under "Login theme", choose "sso-kit-keycloak-theme" and hit Save
- Go to the login page and check if the theme is correctly applied

### To customize your build using `sso-kit-keycloak-theme` as the parent theme:

- Under `themes`, create a folder structure like `your-theme-name/login` and then create these two files:
  - `theme.properties`
  - `resources/login/custom-styles.css`
- On `theme.properties`:
  ```properties
	parent=sso-kit-keycloak-theme

	styles=css/lumo-common.css css/lumo-styles.css css/custom-styles.css
  ```
- On `custom-styles.css`:
  ```css
	/* HERE YOU CAN CUSTOMIZE SOME CSS PROPERTIES LIKE THE PRIMARY COLORS, FOR INSTANCE */
	html {
      --lumo-primary-color: rgb(22, 138, 72);
      --lumo-primary-color-50pct: rgba(22, 138, 72, 0.704);
	}
  ```
- The new theme should be available on the same "Login theme" select mentioned previously

Fixes #18

## Type of change

- [ ] Bugfix
- [X] Feature

### CODE REVIEW

File: sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/login.ftl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    <#elseif section = "form">
+    <div id="kc-form" class="${properties.formContainer}">
+      <div id="kc-form-wrapper">
+        <#if realm.password>
+            <form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="${url.loginAction}" method="post">
+                <#if !usernameHidden??>
+                    <div class="${properties.kcFormGroupClass!}">
+                        <label for="username" class="${properties.kcLabelClass!}"><#if !realm.loginWithEmailAllowed>${msg("username")}<#elseif !realm.registrationEmailAsUsername>${msg("usernameOrEmail")}<#else>${msg("email")}</#if></label>
+
+												<div class="${properties.kcInputContainer}">
```
Comment (heruan) [inline sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/login.ftl:14]: Starting from this line indentation seems to go off 🤔

**Review by heruan — CHANGES_REQUESTED** [review]

File: sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/login.ftl (hunk 1 above)
Comment (DiegoCardoso) [reply sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/login.ftl]: Reformatted the code.

File: sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/template.ftl [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        </#if>
+    </head>
+
+    <body class="${properties.kcBodyClass!}">
+    <div class="${properties.kcLoginClass!}">
+        <div id="kc-header" class="${properties.kcHeaderClass!}">
+            <div id="kc-header-wrapper"
+                 class="${properties.kcHeaderWrapperClass!}">
+                ${kcSanitize(msg("loginTitleHtml",(realm.displayNameHtml!'')))?no_esc}
+            </div>
```
Comment (heruan) [inline sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/template.ftl:48]: If `realm.displayNameHtml` is not defined, can we have a default `<h1>` with `realm.displayName` here?

**Review by heruan — CHANGES_REQUESTED** [review]

File: sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/template.ftl (hunk 2 above)
Comment (DiegoCardoso) [reply sso-kit-keycloak-theme/src/main/resources/theme/sso-kit-keycloak-theme/login/template.ftl]: Done. I noticed that `realm.displayNameHtml` defaults to `realm.displayName` when nothing is provided at the "HTML Display name" field on Keycloak, so the way I figure to check when a custom markup is provided is to test when those two properties are different. When no `realm.displayNameHtml` is provided, the `realm.displayName` is shown inside a `<h1>` tag.

File: sso-kit-starter/sso-kit-starter.iml [hunk 3]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
```
Comment (heruan) [inline sso-kit-starter/sso-kit-starter.iml:2]: I suppose this file slipped into the commit by mistake 🙂

Comment (DiegoCardoso) [reply sso-kit-starter/sso-kit-starter.iml]: Yeah, totally 😅

**Comment by DiegoCardoso** [comment]
> Looks good to me! I would only make a structural change: place this module under an intermediate `sso-kit-keycloak` parent module and rename it to `sso-kit-keycloak-lumo`. This way we'll have place to add additional Keycloak specific modules and new themes!

Refactored.

**Review by heruan — CHANGES_REQUESTED** [review]
Let's also rename the module to `sso-kit-keycloak-lumo` so we can later add other themes, e.g. Material.

**Comment by DiegoCardoso** [comment]
> Let's also rename the module to `sso-kit-keycloak-lumo` so we can later add other themes, e.g. Material.

My bad. I made the changes a while ago but forgot to push.

**Review by heruan — APPROVED** [review]

**Pull request merged**